administration/eslint.config.mjs

```diff
@@ -0,0 +1,217 @@
+import { fixupConfigRules, fixupPluginRules } from '@eslint/compat'
+import { FlatCompat } from '@eslint/eslintrc'
+import js from '@eslint/js'
+import typescriptEslint from '@typescript-eslint/eslint-plugin'
+import tsParser from '@typescript-eslint/parser'
+import jest from 'eslint-plugin-jest'
+import jsxA11Y from 'eslint-plugin-jsx-a11y'
+import jsxExpressions from 'eslint-plugin-jsx-expressions'
+import preferArrow from 'eslint-plugin-prefer-arrow'
+import react from 'eslint-plugin-react'
+import reactHooks from 'eslint-plugin-react-hooks'
+import { createRequire } from 'module'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+
+const require = createRequire(import.meta.url)
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
```

Comment:
:wrench: We were not using these configs before, was there any reason you added these? I think using `typescript-eslint` recommended config provides better defaults (including useful rules for typescript) already, so these should not be necessary (https://eslint.org/docs/latest/use/migrate-to-9.0.0#-eslintrecommended-and-eslintall-no-longer-accepted-in-flat-config)